File: src/tests/java/page/DevTests.java
package page;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import static org.testng.Assert.assertTrue;

public class DevTests {
    private WebDriver driver;
    private DevHomePage homePage;
    private DevAboutPage aboutPage;

    @BeforeMethod
    public void setUp() {
        driver = new ChromeDriver();
        homePage = new DevHomePage(driver);
        aboutPage = new DevAboutPage(driver);

        homePage.open();
    }

    @Test
    public void userSelectsAboutLink_GetsRedirectedToAboutPage() {
        homePage.goToAboutPage();

        assertTrue(aboutPage.isVisible());
    }

    @Test
    public void userSearchesForTest_ReturnsAllTestRelatedPosts() {
        homePage.searchFor("test");

        assertTrue(homePage.hasArticles());
    }

    @AfterMethod
    public void tearDown() {
        driver.quit();
    }
}

File: src/tests/java/page/DevTests.kt
package page

import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver
import org.testng.Assert.assertTrue
import org.testng.annotations.AfterMethod
import org.testng.annotations.BeforeMethod
import org.testng.annotations.Test

class DevTests {
    private lateinit var driver: WebDriver
    private lateinit var homePage: DevHomePage
    private lateinit var aboutPage: DevAboutPage

    @BeforeMethod
    fun setUp() {
        driver = ChromeDriver()
        homePage = DevHomePage(driver)
        aboutPage = DevAboutPage(driver)

        homePage.open()
    }

    @Test
    fun goToAboutPage_ClickAboutLinkFromHomePage_RedirectsUserToAboutPage() {
        homePage.goToAboutPage()
        assertTrue(aboutPage.isAtAboutPage)
    }

    @Test
    fun searchFor_QueryingForTest_ReturnsAllPostsRelatedToTest() {
        homePage.searchFor("test")
        assertTrue(homePage.hasArticles)
    }

    @AfterMethod
    fun tearDown() {
        driver.quit()
    }
}
